Sources/securekeystore/KeychainManager/KeychainManagerImpl.swift

```diff
@@ -0,0 +1,190 @@
+import Security
+import Foundation
+
+class KeychainManagerImpl: KeychainManagerProtocol {
+    
+    private let biometrics: BiometricsProtocol
+    private let keychainQueue = DispatchQueue(label: "keychainQueue")
+
+    init(biometrics: BiometricsProtocol) {
+        self.biometrics = biometrics
+    }
+    
+    private func authenticateUser(completion: @escaping (Bool) -> Void) {
+        biometrics.authenticateUser(reason: "Authenticate to access your keys") { success, error in
+            if let error = error {
+                print("Biometric authentication failed: \(error.localizedDescription)")
+            }
+            completion(success)
+        }
+    }
+    
+    func storeKey(key: SecKey, tag: String, completion: @escaping (Bool) -> Void) {
+        keychainQueue.async {
+            self.authenticateUser { success in
+                guard success else {
+                    completion(false)
+                    return
+                }
+
+                let query: [String: Any] = [
+                    kSecClass as String: kSecClassKey,
+                    kSecAttrLabel as String: tag+"label",
+                    kSecAttrApplicationTag as String: tag,
+                    kSecValueRef as String: key
+                ]
+                
+                let status = SecItemAdd(query as NSDictionary, nil)
+                completion(status == errSecSuccess)
+            }
+        }
+    }
+    func dumpAllSecKeys() {
+        let query: [String: Any] = [
+            kSecClass as String: kSecClassGenericPassword,
+            kSecMatchLimit as String: kSecMatchLimitAll,
+        ]
+        
+        var result: AnyObject?
+        let status = SecItemCopyMatching(query as NSDictionary, &result)
+        
+        guard status == errSecSuccess else {
+            print("Failed to retrieve keys from keychain: \(status)")
+            return
+        }
+        
+        print("key dumpppp",result ?? [])
```

Comment:
why do we need this log?

Reply:
Not needed, added for debugging purposeless. Removing 😅